rules/opentitan_gdb_test.bzl

```diff
@@ -111,3 +115,24 @@ opentitan_gdb_fpga_cw310_test = rv_rule(
     },
     test = True,
 )
+
+# Orchestrate opentitantool, OpenOCD, and GDB to load the given program into
+# SRAM and execute it in-place. This macro assumes that a CW310 FPGA and an
+# ARM-USB-TINY-H JTAG debugger are attached to the host.
+def opentitan_gdb_fpga_cw310_test(
+        tags = [],
+        **kwargs):
+    _opentitan_gdb_fpga_cw310_test(
+        tags = tags + [
+            "exclusive",  # Prevent FPGA tests from running concurrently.
```

Comment:
can we also have "jtag" and cw310_test_rom, cw310_rom or cw310_rom_variant if cached or uncached bitstreams are used?

Reply:
Let's see, adding "jtag" unconditionally would be trivial.

I could automatically determine `cw310_test_rom` and `cw310_rom`, but I'm not sure how to detect whether we should add `cw310_rom_variant` or the theoretical `cw310_test_rom_variant`. Maybe we can attempt to infer the tag and barf when we can't. WDYT about something like this?

```starlark

cw310_tag_count = len(t for t in tags if t in ['cw310_rom', 'cw310_test_rom', 'cw310_rom_variant', 'cw310_test_rom_variant'])

if cw310_tag_count > 1:
  fail("Saw conflicting cw310_* tags for {}".format(name))
elif cw310_tag_count == 0:
  if bitstream == '//hw/bitstream:rom':
    tags.append('cw310_rom')
  elif bitstream == '/hw/bitstream:test_rom':
    tags.append('cw310_test_rom')
  else:
    fail("Failed to infer a 'cw310_*' tag for {}.".format(name))
```

Reply:
I'd rather it be closer to the functest's strategy of having the BUILD file enumerate targets and then making the test for each of them and infering the tag under the hood. Until then they can set a bitstream and get yelled at by #16010 until the tag is correct.

Reply:
Above you're right that we depend on the cw310 tag, and it should probably aslso be here as it's in the functests.

Reply:
OK, I'm unconditionally adding "jtag" and "cw310" tags in this macro. The targets that use opentitan_gdb_fpga_cw310_test() all define "cw310_rom" because they use the ROM (even though it's a splice of //hw/bitstream:rom).